edb/tools/profiling/profiler.py

```diff
@@ -268,6 +272,39 @@ def accumulate_singledispatch_traces(self) -> Dict[FunctionID, CallCounts]:
         return result
 
 
+def profile_memory(func: Callable[[], Any]) -> MemoryFrame:
+    """Profile memory and return a tree of statistics.
+
+    Feed those to `render_memory_svg()` to write an SVG.
+    """
+    import tracemalloc
+
+    tracemalloc.start(1024)
```

Comment:
What process did you use to arrive at this particular number of frames?

Reply:
1. I wanted a number that was the next perfect square after a safe number that covered all recursion in our benchmarks.
2. 1000 is the recursion limit.

It's an arbitrary number but I wanted the flame graph to be able to show all allocations.

Reply:
:+1: